### ISSUE TITLE
Any Desktop Icon with Terminal=true will not open a terminal sesssion.

### ISSUE THREAD

**Issue by aversatech** [linuxmint/cinnamon#8268]
```
 * Cinnamon version (cinnamon --version)
 * Distribution Mint 19.1
 * Nvidia 1070 using the latest linux driver.
 * 64 bit
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
 ```

**Issue**
Any .desktop Icon with a entry under [Desktop Entry] of Terminal=true will not open a terminal session.
This worked up until Mint 19 was upgraded to 19.1

**Steps to reproduce**
I booted Mint 19.1 from DVD and created a simple bash script called test.sh to echo a line and a test.desktop icon to call it. I still see no terminal session with Terminal=true or Terminal=1. If I use Exec=gnome-terminal -- test.sh then it works correctly. 

**Expected behaviour**
A terminal session should start, when using the Terminal=true entry.

**Other information**

**Comment by icarter09** [comment]
Same issue as #8246. The value of `exec=gnome-terminal` is never altered so no matter what Preferred Terminal Application is set or custom set, that new value never replaces `gnome-terminal`.

**Comment by goebbe** [comment]
This is a regression. 
It worked on Mint 19 - it does not work on Mint 19.1.
I ran into this bug after updating to 19.1.   The launcher on the desktop did not work any more.

Side observation:  
When I created the launcher (in my old Mint 19)  I opted for also including the launcher into the Mint-menu. The launcher in the menu still works, after the update to Mint 19.1 !! Only the launcher on the desktop stopped working. 
Workaround: Manually add "gnome-terminal -- " in front of the command, on the desktop launcher did the trick for me.

**Comment by astier** [comment]
I have tried it neither on Linux Mint nor in Cinnamon but hard-linking the desired terminal to xterm solves the issue for me. I think this is a deeper issue which is not caused by Cinnamon. After searching on the Internet I found people from other distributions with similar issues. The hard-link approach was the only one which worked for me. For the st terminal it would be:

```sh
sudo ln -f /usr/local/bin/st /usr/bin/xterm
```
